src/Storefront/Framework/Routing/Router.php

```diff
@@ -194,6 +194,11 @@ private function getBasePath(): string
 
     private function isStorefrontRoute(string $name): bool
     {
+        /** Exception for @see \Shopware\Core\Checkout\Payment\Controller\PaymentController::finalizeTransaction */
+        if (str_starts_with($name, 'payment.')) {
```

Comment:
why not that one specific?

Reply:
and maybe a storefront framework config:

```
storefront:
  router:
    whistelisted:
       - payment.finalize.transaction
```

Reply:
sounds good :+1: 

also this made me think, that changing the whole check would be a breaking change :grimacing: